shader-nodes/shader-nodes/cross.js

```diff
@@ -0,0 +1,37 @@
+// node
+var cross = {
+    code: "vec3 cross(in vec3 a, in vec3 b)\n{\n    return cross(a,b);\n}",
+    // placeholder meta data - structure will be finalized in MVP S3
+    meta: {
+        ports: [
+            { id: 0 },
+            { id: 1 },
+            { id: 2 }
+        ]
+    },
+    // placeholder editor data - structure will be finalized in MVP S3
+    editor: {
+        label: "CROSS PRODUCT",
+        ports: [
+            { label: '' },
+            { label: 'TEXTURE' },
+            { label: 'UV' }
+        ]
+    }
+};
+
+// generator
+cross.gen = function ( argTypes, options ) {
+    var code = this.code;
+
+    if (options && options.precision) {
+        // precision - alter name to create variant
+        code = code.replace('vec3 cross', 'vec3 cross_' + options.precision);
+        // precision - tmp: useful comment
+        code = code.replace( '{\n', '{\n// precision ' + options.precision + ' float;\n');
+    }
```

Comment:
I think you should try something more elegant than starting with a code string and then applying some hard to read string.replace calls.

Maybe even template strings directly (see https://hacks.mozilla.org/2015/05/es6-in-depth-template-strings-2/) - you'd need to merge master to your branch to get ES6 support. Or something along similar format to allow easy replacements are runtime, and good readability.

(adding as a screenshot to get better highlighting)
<img width="370" alt="Screen Shot 2020-12-31 at 2 29 53 PM" src="https://user-images.githubusercontent.com/59932779/103414165-aca95480-4b74-11eb-8d61-1b4f44e069e2.png">

Reply:
ooh nice - I'll merge master and use this - we can also improve a lot of the shader chunk stuff in the generator

Reply:
I'm planning to look into this - but is it necessary for this PR? This PR isn't supposed to contain the final list of core nodes or core node generators - just the underlying systems to allow for them to be developed (alongside the MVP graph editor). In fact it would be great (and probably necessary) if core nodes could be developed by several developers and eventually by external contributors

Reply:
My opinion is that any code that makes it into code base needs to be a good quality, as we wont be reviewing it again. I agree all core nodes can be developed by multiple programmers at a later stage, but here we're looking at a system that allows us to do it effectively. But I understand that it's not easy to develop everything in a larger PR to best quality, and so I'll call @willeastcott for an opinion here as well.

Reply:
IMO replace() isn't low quality code - it is what was possible without newer es6 language features - and is used in the engine in the shader generators for all material texture map UVs.

Reply:
I've replaced `replace()` (and some other string generation) with string templates - it does increase readability a bit.